synapse/app/_base.py

```diff
@@ -292,6 +302,25 @@ def setup_sentry(hs):
         scope.set_tag("worker_name", name)
 
 
+def setup_sdnotify(hs):
+    """Adds process state hooks to tell systemd what we are up to.
+    """
+
+    # Tell systemd our state, if we're using it. This will silently fail if
+    # we're not using systemd.
+    sd_channel = sdnotify.SystemdNotifier()
+
+    hs.get_reactor().addSystemEventTrigger(
+        "after",
+        "startup",
+        lambda: sd_channel.notify("READY=1\nMAINPID=%s" % (os.getpid())),
```

Comment:
doesn't this need to happen a bit later? IIRC we do some acme foo after the reactor starts but before we bind the http and replication listeners

Reply:
Oh, it looked like we started listening pretty much immediately we start the reactor, but actually we do some acme stuff first. Good spot.